component/portal/src/main/java/org/exoplatform/portal/rest/GroupRestResourcesV1.java

```diff
@@ -580,6 +584,67 @@ public Response deleteMembership(@ApiParam(
     return Response.noContent().build();
   }
 
+  @GET
+  @Path("treeMembers")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @ApiOperation(
+          value = "Gets groups tree",
+          httpMethod = "GET",
+          response = Response.class,
+          notes = "This returns the list of groups children containing the given search text for the current user"
+  )
+  @ApiResponses(
+          value = {
+                  @ApiResponse(code = 200, message = "Request fulfilled"),
+                  @ApiResponse(code = 401, message = "User not authorized to call this endpoint"),
+                  @ApiResponse(code = 500, message = "Internal server error") }
+  )
+  public Response getGroupsTreeMembers(@Context UriInfo uriInfo,
+                                       @ApiParam(value = "Search text to filter groups") @QueryParam(
+                                               "q"
+                                       ) String q,
+                                       @ApiParam(value = "Offset") @QueryParam(
+                                               "offset"
+                                       ) int offset,
+                                       @ApiParam(value = "Limit") @QueryParam(
+                                               "limit"
+                                       ) int limit,
+                                       @QueryParam(
+                                               "returnSize"
+                                       ) boolean returnSize) throws Exception {
+
+    Identity identity = ConversationState.getCurrent().getIdentity();
+    if (!userACL.isSuperUser() && !userACL.isUserInGroup(userACL.getAdminGroups())
+            && !identity.isMemberOf(PROCESSES_GROUP)) {
+      return Response.status(Response.Status.UNAUTHORIZED).build();
+    }
+    offset = offset > 0 ? offset : DEFAULT_OFFSET;
+    limit = limit > 0 ? limit : DEFAULT_LIMIT;
+
+    Collection<Group> UserGroupsList = organizationService.getGroupHandler().findGroupsOfUserByKeyword(identity.getUserId(), q);
```

Comment:
injection risk. You're taking input from params without any sanitization/escaping.